cpp/src/arrow/dataset/partition_test.cc

```diff
@@ -322,8 +397,9 @@ TEST_F(TestPartitioning, DiscoverHiveSchema) {
 }
 
```

Comment:
There was already one in dictionary inference but I added one in the non-dictionary inference case too.  This also prompted me to consider the case `{"/alpha=_FALLBACK_STRING_"}` which I've changed to be an error instead of returning an empty schema.  This matches directory partitioning.